docs/sbft-formal-model/proof.dfy

```diff
@@ -434,24 +434,24 @@ module Proof {
     ProofCommitMsgsFromHonestSendersAgree(c, v, v', step);
   }
 
-  lemma QuorumOfPreparesInNetworkMonotonic(c: Constants, v:Variables, v':Variables, step:Step, h_step:Replica.Step)
-    requires NextStep(c, v, v', step)
-    requires c.clusterConfig.IsReplica(step.id)
-    requires var h_c := c.hosts[step.id].replicaConstants;
-             var h_v := v.hosts[step.id].replicaVariables;
-             var h_v' := v'.hosts[step.id].replicaVariables;
-             Replica.NextStep(h_c, h_v, h_v', step.msgOps, h_step)
-    ensures (forall view, seqID, clientOp | QuorumOfPreparesInNetwork(c, v, view, seqID, clientOp)
-                                        :: QuorumOfPreparesInNetwork(c, v', view, seqID, clientOp))
-  {
-    forall view, seqID, clientOp | QuorumOfPreparesInNetwork(c, v, view, seqID, clientOp)
-                                  ensures QuorumOfPreparesInNetwork(c, v', view, seqID, clientOp)
-    {
-      var senders := Messages.sendersOf(sentPreparesForSeqID(c, v, view, seqID, clientOp));
-      var senders' := Messages.sendersOf(sentPreparesForSeqID(c, v', view, seqID, clientOp));
-      Library.SubsetCardinality(senders, senders');
-    }
-  }
+  // lemma QuorumOfPreparesInNetworkMonotonic(c: Constants, v:Variables, v':Variables, step:Step, h_step:Replica.Step)
+  //   requires NextStep(c, v, v', step)
+  //   requires c.clusterConfig.IsReplica(step.id)
+  //   requires var h_c := c.hosts[step.id].replicaConstants;
+  //            var h_v := v.hosts[step.id].replicaVariables;
+  //            var h_v' := v'.hosts[step.id].replicaVariables;
+  //            Replica.NextStep(h_c, h_v, h_v', step.msgOps, h_step)
+  //   ensures (forall view, seqID, clientOp | QuorumOfPreparesInNetwork(c, v, view, seqID, clientOp)
+  //                                       :: QuorumOfPreparesInNetwork(c, v', view, seqID, clientOp))
+  // {
+  //   forall view, seqID, clientOp | QuorumOfPreparesInNetwork(c, v, view, seqID, clientOp)
+  //                                 ensures QuorumOfPreparesInNetwork(c, v', view, seqID, clientOp)
+  //   {
+  //     var senders := Messages.sendersOf(sentPreparesForSeqID(c, v, view, seqID, clientOp));
+  //     var senders' := Messages.sendersOf(sentPreparesForSeqID(c, v', view, seqID, clientOp));
+  //     Library.SubsetCardinality(senders, senders');
+  //   }
+  // }
```

Comment:
why not delete this lemma, instead of commenting out?

Reply:
It was very useful during the initial phase of the proof and we left it for future debugging purposes. I will fix it to take in consideration the FaultyReplica's steps.